### PR TITLE
LPD-18967 Layout and DraftLayout mapping is wrong after exporting/importing a site

### PR DESCRIPTION
**Pull request by DaniSzimko**
Issue : https://liferay.atlassian.net/browse/LPD-18967

I switched the layouts Map's key to Plid, since the Plid is unique and the layoutId is not.

### CODE REVIEW

**Comment by DaniSzimko** [comment]
ci:test:headless-staging

**Comment by DaniSzimko** [comment]
ci:test:relevant

**Comment by DaniSzimko** [comment]
ci:test:sf

**Comment by DaniSzimko** [comment]
ci:test:relevant

**Comment by DaniSzimko** [comment]
ci:test:headless-staging

**Comment by DaniSzimko** [comment]
ci:test:stable

**Comment by DaniSzimko** [comment]
ci:test:sf

**Comment by DaniSzimko** [comment]
ci:test:sf

**Comment by DaniSzimko** [comment]
ci:test:sf

**Comment by DaniSzimko** [comment]
ci:test:relevant

**Comment by DaniSzimko** [comment]
ci:test:headless-staging

File: modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Daniel Szimko
+ */
+@RunWith(Arquillian.class)
+public class LayoutFromMasterPageTemplateExportImportTest {
```
Comment (4lejandrito) [inline modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java:64]: I would move this back to `LayoutExportImportTest`. We normally group test by functionality (LayoutExportImport)  and then have specific tests for each case (like this one). I am pretty sure Brian would ask for the same.

File: modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	public void importLayouts(
```
Comment (4lejandrito) [inline modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java:71]: The only public methods in a test should be `setUp` , `tearDown` and the `@Test`s themselves.

File: modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+			importedLayout4.getName(), importedDraftLayout4.getName());
+
+		Assert.assertTrue(
+			importedDraftLayoutOfMasterPageTemplate.isDraftLayout());
+		Assert.assertEquals(
+			importedMasterPageTemplateLayout.getName(),
+			importedDraftLayoutOfMasterPageTemplate.getName());
+	}
+
+	protected void exportImportLayouts(
```
Comment (4lejandrito) [inline modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java:244]: All these should be private methods. Protected methods should only be present in BaseTest classes that are meant to be extended.

File: modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+		UserTestUtil.setUser(TestPropsValues.getUser());
+	}
+
+	@Test
+	public void testExportImportLayoutFromMasterTemplateAndDraftLayoutMappingOnImportSide()
+		throws Exception {
+
+		// make sure that the layout ID is the same as layout4's layoutId
+
+		LayoutTestUtil.addTypePortletLayout(_group, true);
```
Comment (4lejandrito) [inline modules/apps/export-import/export-import-test/src/testIntegration/java/com/liferay/exportimport/test/LayoutFromMasterPageTemplateExportImportTest.java:112]: Is it necessary to create 5 to reproduce the issue? Can't we just simplify and create the minimum possible to reproduce it?

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by DaniSzimko** [comment]
ci:test:relevant

**Comment by DaniSzimko** [comment]
ci:test:headless-staging

**Comment by DaniSzimko** [comment]
Hello @4lejandrito , I added the requested changes.

**Comment by DaniSzimko** [comment]
ci:test:relevant

File: modules/apps/export-import/export-import-test/build.gradle [hunk 5]
```diff
@@ -2,6 +2,7 @@ dependencies {
 	testIntegrationImplementation group: "com.liferay.portal", name: "com.liferay.portal.impl", version: "default"
 	testIntegrationImplementation group: "javax.portlet", name: "portlet-api", version: "3.0.1"
 	testIntegrationImplementation group: "org.apache.felix", name: "org.apache.felix.http.servlet-api", version: "1.1.2"
+	testIntegrationImplementation group: "org.mockito", name: "mockito-core", version: "4.5.1"
```
Comment (4lejandrito) [inline modules/apps/export-import/export-import-test/build.gradle:5]: This is not needed.

File: modules/apps/export-import/export-import-test/build.gradle [hunk 6]
```diff
@@ -11,13 +12,17 @@ dependencies {
 	testIntegrationImplementation project(":apps:dynamic-data-mapping:dynamic-data-mapping-test-util")
 	testIntegrationImplementation project(":apps:export-import:export-import-api")
 	testIntegrationImplementation project(":apps:export-import:export-import-test-util")
+	testIntegrationImplementation project(":apps:fragment:fragment-api")
+	testIntegrationImplementation project(":apps:fragment:fragment-entry-processor:fragment-entry-processor-drop-zone")
```
Comment (4lejandrito) [inline modules/apps/export-import/export-import-test/build.gradle:16]: This is not needed.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by DaniSzimko** [comment]
ci:test:sf

**Comment by DaniSzimko** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
Sent to echo for review https://github.com/liferay-echo/liferay-portal/pull/15364.

**Pull request closed without merging**